src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public void addFruit(String fruit, int quantity) {
+        storage.put(fruit, storage.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public void removeFruit(String fruit, int quantity) {
+        int currentQuantity = storage.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Insufficient stock for " + fruit);
+        }
+        storage.put(fruit, currentQuantity - quantity);
+    }
+
+    public int getQuantity(String fruit) {
+        return storage.getOrDefault(fruit, 0);
+    }
+
+    public void updateFruit(String fruit, int quantity) {
+        storage.put(fruit, quantity);
+    }
+
+    public Map<String, Integer> getStorage() {
+        return new HashMap<>(storage);
+    }

```

Comment:
Returning a new HashMap copy of the storage might be unnecessary and can lead to performance issues if the storage is large. If the intention is to prevent the caller from modifying the internal map, consider returning an unmodifiable view of the map using Collections.unmodifiableMap(storage).